src/main/java/com/tabnine/Initializer.java

```diff
@@ -43,6 +44,7 @@ private void initialize() {
       CapabilitiesService.getInstance().init();
       TabnineUpdater.pollUpdates();
       PluginInstaller.addStateListener(instanceOfUninstallListener());
+      UserState.init();
```

Comment:
weren't we using `StateRequest` anywhere else? how aobut migrating existing places to use this singleton state?

Reply:
I'm not sure that a stale state fits for the other usages. for me it's ok as I'm using only the installation time.